//  File: tao_child_channel.proto
//  Author: Tom Roeder <tmroeder@google.com>
//
//  Description: Protocol buffers for TaoChildChannel RPC.
//
//  Copyright (c) 2013, Google Inc.  All rights reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package tao;

enum TaoChildRequestType {
  TAO_CHILD_RPC_UNKNOWN = 0;  // unused
  TAO_CHILD_RPC_GET_RANDOM_BYTES = 1;
  TAO_CHILD_RPC_SEAL = 2;
  TAO_CHILD_RPC_UNSEAL = 3;
  TAO_CHILD_RPC_ATTEST = 4;
  TAO_CHILD_RPC_GET_HOSTED_PROGRAM_FULL_NAME = 5;
  TAO_CHILD_RPC_EXTEND_NAME = 6;
}

message TaoChildRequest {
  required TaoChildRequestType rpc = 1;
  optional bytes data = 2;
  optional int32 size = 3;
  optional int32 policy = 4;  // TODO(kwalsh) fix policy hack
}

message TaoChildResponse {
  required TaoChildRequestType rpc = 1;
  required bool success = 2;
  optional bytes data = 3;
  optional int32 policy = 4;  // TODO(kwalsh) fix policy hack
}
